src/cpp/rtps/flowcontrol/FlowControllerFactory.cpp

```diff
@@ -25,26 +25,26 @@ void FlowControllerFactory::init(
                 pure_sync_flow_controller_name,
                 std::unique_ptr<FlowController>(
                     new FlowControllerImpl<FlowControllerPureSyncPublishMode,
-                    FlowControllerFifoSchedule>(participant_, nullptr))));
+                    FlowControllerFifoSchedule>(participant_, nullptr, 0))));
     // SyncFlowController -> used by rest of besteffort writers.
     flow_controllers_.insert(decltype(flow_controllers_)::value_type(
                 sync_flow_controller_name,
                 std::unique_ptr<FlowController>(
                     new FlowControllerImpl<FlowControllerSyncPublishMode,
```

Comment:
```suggestion
                    FlowControllerFifoSchedule>(participant_, nullptr, async_controller_index_++))));
    // SyncFlowController -> used by rest of besteffort writers.
    flow_controllers_.insert(decltype(flow_controllers_)::value_type(
                sync_flow_controller_name,
                std::unique_ptr<FlowController>(
                    new FlowControllerImpl<FlowControllerSyncPublishMode,
```

If `async_controller_index_` is defined as 0 by default, will the first and second inserted flow controllers share the `0` index? I think so.

Reply:
The first one will not create a thread, so we don't mind its id 😉